script.js

```diff
@@ -6,4 +6,62 @@ document.querySelector(".trigger").addEventListener("click", e => {
 
 document.querySelector(".close-button").addEventListener("click", e => {
     modal.classList.toggle("show-modal");
-});
\ No newline at end of file
+});
+
+function modalEndGame () {
+    let modalEndGame = document.querySelector('#endModal')
+        modalEndGame.classList.toggle("blockEndGame")
+}
+
+const display = document.querySelector('.display')
+let colorPattern = []
+let timesRun = 1
+function lightUpSquare() {
+    let colours = ["blue", "green", "yellow", "red"]
+    let random_colour = (colours[Math.floor(Math.random()*colours.length)])
+    colorPattern.push(random_colour)
+    document.querySelector('#' + random_colour).classList.add('activated')
+    display.classList.add('unclickable');
+    setTimeout(() => {
+        removeLight(random_colour)
+    }, 500)
+    if (timesRun < 4) {
+        timesRun++
+        setTimeout(lightUpSquare, 1000)
+    } else {
+        display.classList.remove('unclickable');
+        playerTurn()
+    }
+}
+
+function removeLight(random_colour) {
+    document.querySelector('#' + random_colour).classList.remove('activated')
+}
+
+const startGame = document.querySelector('.start-button');
```

Comment:
can we group the functions together and the executing code together, at the moment is very messy

Reply:
also this variable feels unnecessary, your only using it once